### PR TITLE
Add renameShapes transformer

### PR DESCRIPTION
**Pull request by srchase**
Adds a `renamesShapes` transformer that can be used to rename shapes in a projection by configuration through smithy-build.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source/1.0/guides/building-models/build-config.rst [hunk 1]
```diff
@@ -1021,6 +1021,50 @@ but keeps the shape if it has any of the provided tags:
...
+.. list-table::
+    :header-rows: 1
+    :widths: 10 20 70
+
+    * - Property
+      - Type
+      - Description
+    * - renamed
+      - ``Map<shapeId, shapeId>``
+      - The map of shape ids of shapes that will be renamed, from the key,
```
Comment (kstich) [inline docs/source/1.0/guides/building-models/build-config.rst:1041]: This is awkwardly worded.

> The map of :ref:\`shape IDs <shape-id>\` to rename. Each key \`\`shapeId\`\` will be renamed to the value \`\`shapeId\`\`.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        private Map<String, String> renamed;
+
+        /**
+         * Sets the map of `from` shape ids to the `to` shape id values that they shapes
+         * will be renamed to.
+         * @param renamed The map of shapes to rename.
+         */
+        public void setRenamed(Map<String, String> renamed) {
+
```
Comment (kstich) [inline smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java:45]: nit: empty line can be removed.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import software.amazon.smithy.model.transform.ModelTransformer;
+
+/**
+ * {@code renameShapes} updates a model by renaming shapes. When
+ * configuring the transformer, a `renamed` property must be set as a
+ * map with the keys as the `from` shape ids that will be renamed `to`
+ * the shape id values. Any references to a renamed shape will also be
+ * updated.
+ */
+public class RenameShapes extends ConfigurableProjectionTransformer<RenameShapes.Config> {
```
Comment (kstich) [inline smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java:33]: This is missing an SPI entry.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    public static final class Config {
+
+        private Map<String, String> renamed;
+
+        /**
+         * Sets the map of `from` shape ids to the `to` shape id values that they shapes
+         * will be renamed to.
+         * @param renamed The map of shapes to rename.
```
Comment (kstich) [inline smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java:42]: nit: usually empty lines between the description and param/return sections for javadoc.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+         */
+        public void setRenamed(Map<String, String> renamed) {
+
+            this.renamed = renamed;
+        }
+
+        /**
+         * Gets the map of shape ids to be renamed.
+         * @return The map of shapes to rename.
```
Comment (kstich) [inline smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java:51]: nit: usually empty lines between the description and param/return sections for javadoc.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+    @Override
+    public String getName() {
+        return "renameShapes";
+    }
+
+    private Map<ShapeId, ShapeId> getShapeIdsToRename(Config config, Model model) {
+        Map<ShapeId, ShapeId> shapeIdMap = new HashMap<>();
+        for (String fromShape : config.getRenamed().keySet()) {
+            ShapeId fromShapeId = ShapeId.from(fromShape);
```
Comment (kstich) [inline smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java:82]: This breaks down if a relative ID is given. I think that's okay as defaulting a namespace here is a bad idea, but should be tested and documented.

**Review by kstich — CHANGES_REQUESTED** [review]

File: docs/source/1.0/guides/building-models/build-config.rst (hunk 1 above)
Comment (srchase) [reply docs/source/1.0/guides/building-models/build-config.rst]: :+1: Fixed in 281dfb5.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java (hunk 2 above)
Comment (srchase) [reply smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java]: Fixed in 281dfb5.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java (hunk 4 above)
Comment (srchase) [reply smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java:44]: Fixed in 281dfb5.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java (hunk 5 above)
Comment (srchase) [reply smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java:53]: Fixed in 281dfb5.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java (hunk 3 above)
Comment (srchase) [reply smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java:34]: Fixed in 12024e9.

File: smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java (hunk 6 above)
Comment (srchase) [reply smithy-build/src/main/java/software/amazon/smithy/build/transforms/RenameShapes.java]: Addressed in cd5bce0.

**Review by kstich — APPROVED** [review]

**Pull request merged**